src/kernels/gfx942_ConvHipIgemmGroupXdlops_metadata.ktn.model

```diff
@@ -0,0 +1,71 @@
+{
+    "num_tuning_params": {
+        "fwd": 13,
+        "bwd": 14,
+        "wrw": 14        
+    },
+    "decodings": {
+        "tunings": {
+            "0": "0",
+            "1": "256",
+            "2": "64",
+            "3": "128",
+            "4": "64",
+            "5": "128",
+            "6": "32",
+            "7": "256",
+            "8": "128",
+            "9": "64",
+            "10": "32",
+            "11": "256",
+            "12": "32",
+            "13": "4",
+            "14": "16",
+            "15": "8",
+            "16": "Default",
+            "17": "Filter1x1Stride1Pad0",
+            "18": "OddC",
+            "19": "Filter1x1Pad0",
+            "20": "8",
+            "21": "4",
+            "22": "2",
+            "23": "32",
+            "24": "Default",
+            "25": "2",
+            "26": "Filter1x1Stride1Pad0",
```

Comment:
I am under the impression that the network predicts tuning parameters as tokens (the keys in this dictionary) and then those tokens get mapped to actual tuning parameters (the values in this dictionary). If that's the case, I'm not sure I understand what, for example, "26" mapping to "Filter1x1Stride1Pad0" means?

Reply:
Could this be a bug in data processing?

Reply:
It's a tuning parameter. Sort of how language models predict an index within the output vocabulary and they then get tokenized to a string. It's not actually predicting token/values in and of themselves but rather indices that correspond to a given token.

Reply:
For example this is a full CK config:
DeviceGroupedConvFwdMultipleABD_Xdl_CShuffle<256, 128, 128, 16, Default, 32, 32, 2, 2, 4, 4, 4, 1, 1>
The "Default" is a tuning parameter.

Reply:
I see. And to double check, is `Filter1x1Stride1Pad0` a tuning parameter, too?

Reply:
Ah another thing. Both "17" and "26" map to "Filter1x1Stride1Pad0". Is that by design?

Reply:
So, this is a byproduct of a decision made a while back. Instead of just tokenizing all unique values throughout the entire vocabulary I would tokenize all unique values within each timestep, as you can see there are multiple "32"s, "64"s,  etc. The overlap of non-numerical tuning parameters has to do with the fact that the model "ConvHipIgemmXdlops" is trained to produce configs/tuning strings for each solver wrw/bwd/fwd and these solvers have the "type", (non-numerical tuning parameter such as "Filter1x1Stride1Pad0"), located at a different timestep/index within the config string.
An example being:

Forward:
DeviceGroupedConvFwdMultipleABD_Xdl_CShuffle<256, 128, 128, 16, Default, 32, 32, 2, 2, 4, 4, 4, 1, 1>

Backward Data:
DeviceGroupedConvBwdDataMultipleD_Xdl_CShuffle_v1<128, 128, 32, 32, 8, 8, Default, 32, 32, 2, 1, 8, 8, 1, 1>

As you can see the type is in a different position within the config string. Now I could revert this decision and tokenize only unique values that are unique throughout the entire vocabulary, but the improvement in the model would have to be tested empirically. I don't know if it would actually lead to better performance.